diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LambdaParameterOrder.kt

```diff
@@ -0,0 +1,39 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.FUNCTION_TYPE
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.WRONG_FUNCTION_ARGUMENTS_ORDER
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.psi.KtFunction
+
+class LambdaParameterOrder(private val configRules: List<RulesConfig>) : Rule("lambda-parameter-order") {
+
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        if (node.elementType == ElementType.FUN) {
+            checkArguments(node)
+        }
+    }
+
+    private fun checkArguments(node: ASTNode) {
+        val funArguments = (node.psi as KtFunction).valueParameters
+        val sortArguments = funArguments.sortedBy { it.typeReference?.node?.hasChildOfType(FUNCTION_TYPE) }
+        funArguments.forEachIndexed { index, ktParameter ->
+            if (ktParameter != sortArguments[index]) {
+                WRONG_FUNCTION_ARGUMENTS_ORDER.warn(configRules, emitWarn, isFixMode, ktParameter.node.text,
```

Comment:
Please either rephrase warning message or provide function name in freeText. I think it's quite misleading now.